Fix incorrect args passed to timezone.set_hwclock

sacren opened the pull request:
On Gentoo, we should check the value passed in to timezone.set_hwclock
and only allow 'UTC' or 'localtime' for argument. If incorrect value is
passed in, it could subsequently fry the important config file.

There is another closely related issue on Gentoo. The value to set
'clock' variable in the config file is either 'UTC' or 'local'. In case
of localtime arg, we should convert the value of 'localtime' to 'local'
when setting that 'clock' variable.

sacren committed:
Fix incorrect args passed to timezone.set_hwclock

On Gentoo, we should check the value passed in to timezone.set_hwclock
and only allow 'UTC' or 'localtime' for argument. If incorrect value is
passed in, it could subsequently fry the important config file.

There is another closely related issue on Gentoo. The value to set
'clock' variable in the config file is either 'UTC' or 'local'. In case
of localtime arg, we should convert the value of 'localtime' to 'local'
when setting that 'clock' variable.

terminalmage commented:
This probably needs to be backported.

terminalmage commented on salt/modules/timezone.py:
```diff
@@ -410,6 +410,12 @@ def set_hwclock(clock):
+            raise SaltInvocationError(
+                'Only \'UTC\' and \'localtime\' are allowed'
+            )
+        if clock.__eq__('localtime'):
```
Can you explain why you are invoking a dunder method here instead of simply using the ``==`` operator? Especially since invoking the dunder is so much slower:

```python
>>> import timeit
>>> t1 = timeit.Timer('clock == "localtime"', 'clock = "localtime"')
>>> t2 = timeit.Timer('clock.__eq__("localtime")', 'clock = "localtime"')
>>> t1.repeat()
[0.061421871185302734, 0.0285952091217041, 0.026736021041870117]
>>> t2.repeat()
[0.1310901641845703, 0.09536910057067871, 0.09479284286499023]
```

sodium-chloride replied:
@terminalmage Thank you for the nice comment. I appreciate it even more for the numbers.

I chose __eq__() method considering clock is a string object. I'm sorry I overlooked the speed issue and I'll follow up with a patch.

terminalmage replied:
Yes, but the ``__eq__()`` method should almost never be invoked directly, so I was curious why this choice was made.

sacren committed:
Fix speed issue

Erik Johnson pointed out the speed issue with compelling numbers:

>>> import timeit
>>> t1 = timeit.Timer('clock == "localtime"', 'clock = "localtime"')
>>> t2 = timeit.Timer('clock.__eq__("localtime")', 'clock = "localtime"')
>>> t1.repeat()
[0.061421871185302734, 0.0285952091217041, 0.026736021041870117]
>>> t2.repeat()
[0.1310901641845703, 0.09536910057067871, 0.09479284286499023]

I'll absolutely fix it by using the simple '==' operator.

Thank you!
